actions/helm-lint/action.yaml

```diff
@@ -1,31 +1,46 @@
 name: "Lint helm chart"
-description: "Lint all helm charts in the chart directory (default: charts)"
+description: "Lint all helm charts in the charts directory of a repository"
 inputs:
-  lintConfig:
-    description: "The path to the lint configuration file."
+  ref:
+    description: "The ref name to checkout the repository."
+    required: false
+    default: ${{ github.event.repository.default_branch }}
+    type: string
+  lint-config-path:
+    description: "The path to the lint configuration file (For an example see https://github.com/helm/chart-testing/blob/main/pkg/config/test_config.yaml)."
+    required: false
     default: ".github/lint-config.yaml"
+    type: string
+  helm-version:
+    description: "The helm version for setting up poetry."
     required: false
+    default: "v3.4.0"
+    type: string
+
 runs:
   using: "composite"
   steps:
-    - name: Set up Helm
-      uses: azure/setup-helm@v1
+    - name: Check out repository
+      uses: actions/checkout@v2
       with:
-        version: "v3.4.0"
+        ref: "${{ inputs.ref }}"
+        # fetch-depth of 2 needed for comparison of helm charts between commits while linting
+        fetch-depth: 2
```

Comment:
Does this still work? I think this has to be 0 now that we compare between branches

Reply:
Let's say the CI runs for commit E:

```
master: A --> B
               \
      feature:  C --> D --> E
```
We have to fetch the changes since B, i.e. looking 3 commits back.  We cannot anticipate the number of commits on a feature-branch, therefore this steps requires the full history ([= 0](https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches)).